Automattic/jetpack
# MU WPCOM: Port the wpcom block editor nux feature from the ETK

Related https://github.com/Automattic/dotcom-forge/issues/8241, https://github.com/Automattic/wp-calypso/issues/92861

## Proposed changes:

> [!TIP]
> Changes are too large because I copied `wpcom-block-editor-nux`, `dotcom-fse/lib` and `@automattic/tour-kit` into `jetpack-mu-wpcom`
>
> It would be better to follow the Test Instructions below to check whether the functionalities work as expected or not

* Migrates wpcom-block-editor-nux to jetpack-mu-wpcom
* Resolved the `@automattic/tour-kit` dependency as follows:
  * The following packages are outdated so we have to release the new packages to resolve the peer dependencies issue
    * `@automattic/components`
    * `@automattic/tour-kit`
    * `@automattic/viewport-react`
  * However, I found we cannot introduce the `@automattic/components` to the `jetpack-mu-wpcom` because:
    * It requires `@automattic/calypso-analytics` package that uses the `EventEmitter` from the `events` package. I'm unsure whether to introduce the `events` package on the browser side.
    * It requires `i18n-calypso` package, and the issue is the same as above.
    * It requires `@emotion/*` packages that are used by the following components and I don't think it makes sense to introduce the whole `@emotion/*` packages for this purpose.
      * `LoadingPlaceholder`
      * `SiteThumbnail`
      * `UpsellMenuGroup`
  * Next, I found the `@automattic/tour-kit` is only used by the `wpcom-block-editor-nux`, so I decided to move the whole package to `jetpack-mu-wpcom` to get rid of those dependencies
  * Finally, just copied the following components to `jetpack-mu-wpcom` because I think they are not too complicated
      * `PaginationControl`
      * `ClipboardButton`
      * `FormInputCheckbox`
      * `FormLabel`
* Resolved the `recordTracksEvent` call
    * Replaced the `recordTracksEvent` with the `wpcomTrackEvent` as we did for other migrations
* Got rid of the type definition to remove the following dependencies
  * `@automattic/onboarding`
  * `starter-page-templates/store`
  * `wpcom-block-editor-nav-sidebar/src/store`

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:

Before getting started, you have to
* Apply changes to your sites
* Sandbox your API and ensure your siteSlug domain points to your sandbox for simple sites

### Welcome Tour

* Go to the Editor
* Make sure the Welcome Tour works as before
  <img width="300" src="https://github.com/user-attachments/assets/46bceea7-3286-46d6-907c-56acf44b3e06" />

### Blogging prompts modal

* Go to /wp-admin/post-new.php?new_prompt=true
* Make sure you can see the blogging prompts modal
  <img width="300" src="https://github.com/user-attachments/assets/1c0b8c6a-5a0e-450a-8948-13da43d5e03d" />

### Purchase notice (Payments block)

* Go to /wp-admin/post-new.php?notice=purchase-success
* Make sure you can see the purchase notice
  <img width="300" src="https://github.com/user-attachments/assets/5b7b199d-e0f8-4ab4-9788-3e04ffe88859" />

### Draft post modal & First post published modal - Only for the "write" intent

* Create a new site
* Select the "Write & Publish" goal
* Select the "Choose a design" starting point
* Verify write your post modal show
  <img width="300" src="https://github.com/user-attachments/assets/0d2f0165-2b8b-4cbe-91a3-a20ce18aaa38" />
* Publish the post
* Verify the first post published modal show (the "write" intent only)
  <img width="300" src="https://github.com/user-attachments/assets/689fc916-8935-4452-823a-4159b8b68c50" />

### [Seller celebration modal](https://github.com/Automattic/wp-calypso/pull/61222) - Only for the "sell" intent

* Go to /setup?siteSlug=your_site
* Select the `Sell online`
* After the onboarding, go to create a new page
* Insert any payment blocks and save
  * Paid Content
  * Pay with PayPal
  * Payment Buttons
* Make sure you can see the seller celebration modal
  <img width="300" src="https://github.com/user-attachments/assets/d6ee5b48-a4b5-4444-8f7d-c9ac5ca8836d" />

### [Sharing modal](https://github.com/Automattic/wp-calypso/pull/76267) - Not for the `start-writing` or `design-first` intent

* Go to /wp-admin/post-new.php to create a new post on your **PUBILC** site
* Publish the post
* Make sure you can see the sharing modal
  <img width="300" src="https://github.com/user-attachments/assets/c1ff7382-a2aa-4ff8-a146-3a1503739a59" />

### [Video celebration modal](https://github.com/Automattic/wp-calypso/pull/67539) - Only for the "videopress" intent

* Start the flow at /setup/videopress
* When you land on the Site Editor, upload a video and save
* Make sure you can see the video celebration modal
  <img width="300" src="https://github.com/user-attachments/assets/19c7b343-f246-4d00-a03d-c307dfc294ce" />
* Note that the modal doesn't seem to work on atomic sites as it requests the `/sites/:site` endpoint to get the value of `options?.launchpad_checklist_tasks_statuses?.video_uploaded` from the response to check whether the video has been uploaded
